api/ingress/v2alpha1/astartedefaultingress_webhook.go

```diff
@@ -207,3 +205,16 @@ func getSecret(c client.Client, secretName string, namespace string, fldPath *fi
 	}
 	return nil
 }
+
+// validateIngressClassAnnotation checks if the Ingress Controller class annotation is valid
+func (r *AstarteDefaultIngress) validateIngressClassAnnotation() *field.Error {
```

Comment:
Not sure about the naming. If I'm not mistaken, we're validating the ingress controller selector. What about `validateIngressControllerSelector`? I think it might help to avoid misunderstandings. If appropriate, let's update also the names in the function implementation.